### PR TITLE
Enable the verse block for mobile

### PR DESCRIPTION
**Pull request by guarani**
Addresses: https://github.com/wordpress-mobile/gutenberg-mobile/issues/1886
Related: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2185

## Description
Enable the verse block for production builds.

## How has this been tested?

1. Download the test build for iOS: https://github.com/wordpress-mobile/WordPress-iOS/pull/14004
2. Perform the test cases for the Verse block from https://github.com/wordpress-mobile/test-cases/pull/28
   - TC001: Tapping return key creates a new line within the block
   - TC002: Setting text alignment works as expected 
   - TC003: Whitespace is preserved when applying formatting options (bold, italics, etc)
   - TC004: The verse block can be merged with the paragraph block
3. Create and publish a verse block on mobile and ensure it's viewable on the web and on the user's site itself
4. Create and publish a verse block on the web and ensure it appears in the app
5. Repeat steps 1 to 4 for the test build for Android: https://github.com/wordpress-mobile/WordPress-Android/pull/11763

## Demo 

| | iOS | Android |
| -- | -- | -- |
| Preview |  |  |

## Types of changes

New feature: verse block for mobile

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/verse/edit.js [hunk 1]
```diff
@@ -34,6 +34,7 @@ export default function VerseEdit( {
 			</BlockControls>
 			<RichText
 				tagName={ Block.pre }
+				identifier='content'
```
Comment (guarani) [inline packages/block-library/src/verse/edit.js:37]: It looks like this is a required attribute to allow the verse block to be merged with the paragraph block:
https://github.com/WordPress/gutenberg/blob/4bbfbc13603d28fcc45368c529954164bf8581de/packages/block-editor/src/store/effects.js#L103
I essentially copied this from the paragraph block:
https://github.com/WordPress/gutenberg/blob/4bbfbc13603d28fcc45368c529954164bf8581de/packages/block-library/src/paragraph/edit.native.js#L39

**Comment by guarani** [comment]
I noticed on the last build that the bug from https://github.com/wordpress-mobile/gutenberg-mobile/issues/1966 is appearing again and I'll need to investigate, so leaving this as a draft.

**Comment by guarani** [comment]
When return is tapped inside a verse block on the mobile editor, a `<br>` tag is inserted – if the same happens on web, a newline character is inserted. 
The post list screen on Android and iOS handle `<br>` tags differently, they are rendered correctly on Android but are not rendered on iOS. (Both platforms handle newline characters correctly.)

| Android | iOS |
| -- | -- |
|  |  |

This issue is not specific to the verse block, – preformatted and pull quote blocks already have this issue – they use `<br>` tags for on mobile but use newline characters on web.

I propose:
- Not treating this as a blocker for releasing the verse block.
- Adding a subissue to the verse block master issue to make the post list screen more resilient on iOS by adding support for rending newline characters. This should fix all preexisting posts that have this issue as well as posts created after the verse block is released.

**Comment by hypest** [comment]
> The post list screen on Android and iOS handle <br> tags differently...

Is this also happening on posts written in the classic editor (Aztec) and on posts written on the web?

Also, is this happening just for PRE-based blocks/markup or the different newline/<br> behavior between the platforms also happens in normal paragraphs? Trying to understand if this is a block editor vs classic editor issue. Or even specific-block vs other blocks issue.

It looks like an issue with the posts list rendering itself, in which case I wouldn't consider it a blocker for the Verse block.

> Adding a subissue to the verse block master issue...

Yes please, thanks!

> Investigate whether new lines should be treated the same on mobile and web for the preformatted and verse blocks, and if possible, unify behavior.

I'd add that as another subtask.

**Comment by guarani** [comment]
> Is this also happening on posts written in the classic editor (Aztec) and on posts written on the web?

This is not happening on posts created in the classic editor nor posts written on the web.

>Also, is this happening just for PRE-based blocks/markup or the different newline/
behavior between the platforms also happens in normal paragraphs?

Tapping return inside a paragraph block just creates a new block, which I think makes paragraphs a different category – do you agree? 

Added issues here:
- https://github.com/wordpress-mobile/gutenberg-mobile/issues/2193 Use newlines characters instead of <br> tags for new lines in pre-based blocks
- https://github.com/wordpress-mobile/WordPress-iOS/issues/14008 Render line breaks properly on the post list screen on iOS

**Comment by etoledom** [comment]
Should we add a Unit Test to this too?
I think it's specially helpful specially in Multiplatform blocks like this one.

Something simple like this should be enough:
https://github.com/WordPress/gutenberg/blob/ea4fa88b4c48d8d5d0de44d5d2eb9cee33a9e424/packages/block-library/src/code/test/edit.native.js#L13-L20

**Comment by guarani** [comment]
Thanks for pointing out the lack of unit tests, @etoledom! I've added those now.

**Comment by guarani** [comment]
I'm getting seeing some tests fail on Travis, e.g.:
- 54638.11 E2E tests (Admin) (2/4)
  - FAIL packages/e2e-tests/specs/editor/blocks/quote.test.js (72.091s)
    - ✕ can be merged into from a paragraph (3236ms)
- 54638.17 E2E tests (Author) (4/4)
  - FAIL packages/e2e-tests/specs/editor/various/rtl.test.js (21.436s)
    - ✕ should navigate inline boundaries (3878ms)

I noticed the build badge was showing that Gutenberg was failing on Friday so I'm not sure if there is some flakiness in the test suite. I've re-run the test suite twice on Travis and I've also ran `npm run test:e2e` locally, but the tests seem to mostly fail and I haven't been able to determine the cause yet.

**Comment by ceyhun** [comment]
All the test cases work great on Android.
On iOS works fine as well except that I found a small issue with making text on second line bold when there's space before it:

**Comment by guarani** [comment]
Thanks a lot for your tests, @ceyhun. 
I double checked that the work to fix the formatting issue is included:
- https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1277
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/2127
It appears that the bug was resolved for the case where formatting is applied to the first line of text in a verse block, but on the 2nd, 3rd, etc, lines the issue is still present.

I also checked the `develop` branch of gutenberg-mobile and this issue _is_ present.

@SergioEstevao I want to let you know about this because you're wrangling the release today and are familiar with this work. I'd like to see if there is still an issue with Aztec that needs to be resolved but I'm not yet very familiar with the repo. 
Do you have any suggestions on a) whether we should still attempt to fix this for the code freeze, b) any pointers about what the problem might be?

**Comment by guarani** [comment]
The bug @ceyhun found was [fixed in the Aztec library by @SergioEstevao](https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1285) which powers the rich text components on mobile, such as that found in the verse block. The gist of it is that the libxml2 library used by Aztec was collapsing whitespace elements when these elements were between other tags.
So in this case:
```
Hello<br>
   <strong>World</strong>
```
the whitespace between the `<br>` and the `<strong>` was being stripped out. Removing the `HTML_PARSE_NOBLANKS` flag on libxml2's configuration fixed this! 🎉

**Comment by guarani** [comment]
## Retest results

I found one new minor bug, and all the other [test cases](https://github.com/WordPress/gutenberg/pull/21883#issue-408783193) passed.

### Minor bug found

If you type bold text, then add a few spaces, then right align the block, the spaces you typed are lost. If no formatting is applied, the bug doesn't occur. It's also iOS specific. I'd say this is another angle of the [bug fixed above](https://github.com/WordPress/gutenberg/pull/21883#issuecomment-627644442) (cc @SergioEstevao). 

@iamthomasbishop, do you agree this is not a blocker for the verse block? If so, I'll add it to the [list of Verse Block non-blockers](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1886#issue-564033295).

Here are two gifs showing the bug. As you can see, only in the first example – where the text is formatted (bolded) – is the bug present

| With formatting ❌ | Without formatting ✅ |
| -- | -- |
|  |  |

**Comment by guarani** [comment]
@ceyhun re-requesting a review here please 🙏.

**Comment by SergioEstevao** [comment]
@guarani I tested with the new version of Aztec and the bug you found when switching alignment is still there. 
The curious thing is that the whitespaces are kept, you can see that by switching to the HTML mode or even by typing a character.

At this point I think this is UITextView layout/presentation issue so it shouldn't be a blocker for this PR.

**Comment by ceyhun** [comment]
> @ceyhun re-requesting a review here please 🙏.

@guarani e2e tests seem to have failed on the gutenberg-mobile PR.

**Review by SergioEstevao — APPROVED** [review]
All working well, extra kudos for adding the unit tests!

**Pull request merged**